src/vm/sampleprofiler.cpp

```diff
@@ -103,34 +120,35 @@ void SampleProfiler::Disable()
         GC_TRIGGERS;
         MODE_ANY;
         // Synchronization of multiple callers occurs in EventPipe::Disable.
-        PRECONDITION(EventPipe::GetLock()->OwnedByCurrentThread());
+        PRECONDITION(EventPipe::IsLockOwnedByCurrentThread());
+        PRECONDITION(s_RefCount >= 0);
```

Comment:
I don't think this precondition is true? If you create one session that does not use the sampling profiler and then disable it DisableInternal() will call here and s_RefCount will be 0.

Reply:
Nevermind, I misread it as s_RefCount > 0. What you wrote looks correct. Ignore silly ole me : )

Reply:
`PRECONDITION(s_RefCount >= 0);` Zero is acceptable :)